blaze/expr/reductions.py

```diff
@@ -14,6 +17,38 @@
 from .expressions import dshape_method_list, method_properties
 
 
+class Window(Expr):
+    __slots__ = (
+        '_hash',
+        '_child', 'kind', 'group_by', 'sort_by', 'preceding', 'following',
+        '_asdshape'
+    )
+
+    __inputs__ = '_child', 'group_by', 'sort_by'
+
+    def __str__(self):
+        slots = 'group_by', 'sort_by', 'preceding', 'following'
+        return '%s(%s, %s)' % (
+            type(self).__name__,
+            self._child,
+            ', '.join('%s=%r' % (slot, getattr(self, slot)) for slot in slots)
+        )
+
+    @property
+    def _inputs(self):
```

Comment:
should we redfine `_inputs` on the superclass to use this? I imagine the case of an input optionally being a literal or expression is probably common. How does `BinOp` handle this?